src/Plugin/resource/DataProvider/DataProviderInterface.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\restful\Plugin\resource\DataProvider\DataProviderInterface.
+ */
+
+namespace Drupal\restful\Plugin\resource\DataProvider;
+
+use Drupal\restful\Plugin\resource\CrudInterface;
+
+interface DataProviderInterface extends CrudInterface {}
```

Comment:
The `DataProviderInterface` **will** have more stuff in the future, not just the `CrudInterface`.

Sorry, there are too many moving pieces. but submitting a complete thing would have been a nightmare to review.